integration-tests/emulated/tests/assets/asset-hub-kusama/src/tests/treasury.rs

```diff
@@ -0,0 +1,271 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Tests concerning the Polkadot Treasury.
+
+use crate::*;
+use emulated_integration_tests_common::accounts::{ALICE, BOB};
+use frame_support::{
+	dispatch::RawOrigin,
+	sp_runtime::traits::Dispatchable,
+	traits::{
+		fungible::Inspect,
+		fungibles::{Create, Inspect as FungiblesInspect, Mutate},
+	},
+};
+use kusama_runtime::OriginCaller;
+use kusama_runtime_constants::currency::GRAND;
+use runtime_common::impls::VersionedLocatableAsset;
+use xcm_executor::traits::ConvertLocation;
+
+// Fund Treasury account on Asset Hub from Treasury account on Relay Chain with KSMs.
+#[test]
+fn spend_ksm_on_asset_hub() {
+	// initial treasury balance on Asset Hub in KSMs.
+	let treasury_balance = 9_000 * GRAND;
+	// the balance spend on Asset Hub.
+	let treasury_spend_balance = 1_000 * GRAND;
+
+	let init_alice_balance = AssetHubKusama::execute_with(|| {
+		<<AssetHubKusama as AssetHubKusamaPallet>::Balances as Inspect<_>>::balance(
+			&AssetHubKusama::account_id_of(ALICE),
+		)
+	});
+
+	Kusama::execute_with(|| {
+		type RuntimeEvent = <Kusama as Chain>::RuntimeEvent;
+		type RuntimeCall = <Kusama as Chain>::RuntimeCall;
+		type Runtime = <Kusama as Chain>::Runtime;
+		type Balances = <Kusama as KusamaPallet>::Balances;
+		type Treasury = <Kusama as KusamaPallet>::Treasury;
+
+		// Fund Treasury account on Asset Hub with KSMs.
+
+		let root = <Kusama as Chain>::RuntimeOrigin::root();
+		let treasury_account = Treasury::account_id();
+
+		// Mint assets to Treasury account on Relay Chain.
+		assert_ok!(Balances::force_set_balance(
+			root.clone(),
+			treasury_account.clone().into(),
+			treasury_balance * 2,
+		));
+
+		let native_asset = Location::here();
+		let asset_hub_location: Location = [Parachain(1000)].into();
+		let treasury_location: Location = (Parent, PalletInstance(18)).into();
+
+		let teleport_call = RuntimeCall::Utility(pallet_utility::Call::<Runtime>::dispatch_as {
+			as_origin: bx!(OriginCaller::system(RawOrigin::Signed(treasury_account))),
+			call: bx!(RuntimeCall::XcmPallet(pallet_xcm::Call::<Runtime>::teleport_assets {
+				dest: bx!(VersionedLocation::V4(asset_hub_location.clone())),
+				beneficiary: bx!(VersionedLocation::V4(treasury_location)),
+				assets: bx!(VersionedAssets::V4(
+					Asset { id: native_asset.clone().into(), fun: treasury_balance.into() }.into()
+				)),
+				fee_asset_item: 0,
+			})),
+		});
+
+		// Dispatched from Root to `despatch_as` `Signed(treasury_account)`.
+		assert_ok!(teleport_call.dispatch(root));
+
+		assert_expected_events!(
+			Kusama,
+			vec![
+				RuntimeEvent::XcmPallet(pallet_xcm::Event::Sent { .. }) => {},
+			]
+		);
+	});
```

Comment:
I think this can be simplified with something like this

```suggestion
        // wtih this
        let relay_treasury_account =
		asset_hub_kusama_runtime::xcm_config::LocationToAccountId::convert_location(
			&treasury_location,
		)
		.unwrap();
        // or even use directly this: asset_hub_kusama_runtime::RelayTreasuryPalletAccount		
		

	AssetHubKusama::fund_accounts(...)
```

Reply:
why you think it needs to be simplified? I want it to be close to what actually will happened on production, and these calls within this test, that what most probably will be proposed for referendum

Reply:
Ok, sure, no problem as it is. I just noticed that this code transfers some balance to the account on AssetHub, and the same can be achieved with simple `AssetHubKusama::fund_accounts(...)`.